Code/Source/VehicleDynamics/ManualControlEventHandler.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "VehicleDynamics/VehicleInputControlBus.h"
+#include <StartingPointInput/InputEventNotificationBus.h>
+
+// TODO - plenty of boilerplate code, seems somewhat redundant since it would be better to be able to map inputs directly
+namespace VehicleDynamics
+{
+    //! A handler for a single input event.
+    class ManualControlSingleEventHandler : private StartingPointInput::InputEventNotificationBus::Handler
+    {
+    public:
+        using OnHeldHandlerFunction = std::function<void(float)>;
+
+        //! Construct the event handler.
+        //! @param eventName which event to handle (e.g. "steering", "accelerate")
+        //! @param handler a function which handles the input, typically through re-publishing it to a vehicle input bus.
+        ManualControlSingleEventHandler(AZStd::string eventName, OnHeldHandlerFunction handler)
```

Comment:
minor: This could be a const ref
```suggestion
        ManualControlSingleEventHandler(const AZStd::string& eventName, OnHeldHandlerFunction handler)
```

Reply:
@spham-amzn Actually the const ref is a pessimization in this case.
At worse by accepting the parameter by value and moving, supplying a lvalue AZStd::string would perform 1 copy and 1 move.

With a `const AZStd::string&`, supplying a string literal would perform 2 copies. One to construct the temporary AZStd::string that binds to a `const AZStd::string&`, the second to then copy into the `m_eventName` member.

Reply:
I guess since it is a constructor that gets called 2 times optimization is not that much of an issue here.
Here, we will likely run into SVO since our strings are short event names: "steering", "accelerate"